vesc_hw_interface/CMakeLists.txt

```diff
@@ -1,69 +1,87 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.8)
 project(vesc_hw_interface)
 
 add_compile_options(-std=c++17)
 
-find_package(
-  catkin REQUIRED
-  COMPONENTS roscpp
-             serial
-             hardware_interface
-             joint_limits_interface
-             controller_manager
-             pluginlib
-             urdf
-             vesc_driver
+if(CMAKE_COMPILER_IS_GNUCXX
+   OR CMAKE_CXX_COMPILER_ID
+      MATCHES
+      "Clang"
 )
+  add_compile_options(
+    -Wall
+    -Wextra
+    -Wpedantic
+  )
+endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS
-    roscpp
-    serial
+set(DEPENDENCIES
+    angles
     hardware_interface
-    joint_limits_interface
-    controller_manager
+    joint_limits
     pluginlib
-    urdf
+    rclcpp
+    rclcpp_lifecycle
+    vesc_msgs
     vesc_driver
 )
 
-include_directories(
-  include
-  ${Boost_INCLUDE_DIRS}
-  ${catkin_INCLUDE_DIRS}
-)
+# find dependencies
+find_package(ament_cmake REQUIRED)
+foreach(Dependency IN ITEMS ${DEPENDENCIES})
+  find_package(${Dependency} REQUIRED)
+endforeach()
+find_package(Threads REQUIRED)
 
 add_library(
-  ${PROJECT_NAME}
+  ${PROJECT_NAME} SHARED
   src/vesc_hw_interface.cpp
   src/vesc_servo_controller.cpp
   src/vesc_wheel_controller.cpp
   src/vesc_step_difference.cpp
 )
-add_dependencies(${PROJECT_NAME} ${catkin_EXPORTED_TARGETS})
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
 
-add_executable(${PROJECT_NAME}_node src/vesc_hw_interface_node.cpp)
-add_dependencies(${PROJECT_NAME}_node ${catkin_EXPORTED_TARGETS})
-target_link_libraries(
-  ${PROJECT_NAME}_node
-  ${catkin_LIBRARIES}
-  ${PROJECT_NAME}
+target_include_directories(${PROJECT_NAME} PUBLIC include)
+
+ament_target_dependencies(${PROJECT_NAME} ${DEPENDENCIES})
```

Comment:
https://github.com/ros-controls/ros2_control_demos/blob/master/example_1/CMakeLists.txt

The demo above add `PUBLIC` here. Is it not required here?

Reply:
I'll change to `PUBLIC` to be able to use from other package.
https://cmake.org/cmake/help/latest/command/target_compile_definitions.html

Reply:
Sorry, I was looking at the wrong line.
`ament_target_dependencies` use `PUBLIC` as default.
https://github.com/ament/ament_cmake/blob/5d3d6e97c8f706861f88ac36a1a730f6487dfec0/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L49